.github/workflows/check-replays.yml

```diff
@@ -0,0 +1,236 @@
+
+          Write-Host "Extracting Archive" -ForegroundColor Cyan
+          & 7z x generals108_gamedata_trimmed.7z -oC:\GameData\Generals
+          Remove-Item generals108_gamedata_trimmed.7z -Verbose
+
+          # Download GeneralsMD (ZH) Game Files
+          # The archive contains these files:
+          # BINKW32.DLL
+          # INIZH.big
+          # MapsZH.big
+          # mss32.dll
+          # W3DZH.big
+          # Data\Scripts\MultiplayerScripts.scb
+          # Data\Scripts\Scripts.ini
+          # Data\Scripts\SkirmishScripts.scb
+
+          Write-Host "Downloading GameData GeneralsMD" -ForegroundColor Cyan
+          aws s3 cp s3://github-ci/zerohour104_gamedata_trimmed.7z zerohour104_gamedata_trimmed.7z --endpoint-url $env:AWS_ENDPOINT_URL
+
+          Write-Host "Verifying File Integrity" -ForegroundColor Cyan
+          $fileHash = (Get-FileHash -Path zerohour104_gamedata_trimmed.7z -Algorithm SHA256).Hash
+          Write-Host "Downloaded file SHA256: $fileHash"
+          Write-Host "Expected file SHA256: $env:EXPECTED_HASH_GENERALSMD"
+          if ($fileHash -ne $env:EXPECTED_HASH_GENERALSMD) {
+              Write-Error "Hash verification failed! File may be corrupted or tampered with."
+              exit 1
+          }
+
+          Write-Host "Extracting Archive" -ForegroundColor Cyan
+          & 7z x zerohour104_gamedata_trimmed.7z -oC:\GameData\GeneralsMD
+          Remove-Item zerohour104_gamedata_trimmed.7z -Verbose
+
+      - name: Set Up Game Data
+        shell: pwsh
+        run: |
+          $source = "C:\GameData\${{ inputs.game }}"
+          $destination = "build"
+          Copy-Item -Path $source\* -Destination $destination -Recurse -Force
+
+      - name: Set Generals InstallPath in Registry
+        shell: pwsh
+        run: |
+          # Zero Hour loads some Generals files and needs this registry key to find the
+          # Generals data files.
+
+          $regPath = "HKCU:\SOFTWARE\Electronic Arts\EA Games\Generals"
+          $installPath = "C:\GameData\Generals\"
+
+          # Ensure the key exists
+          if (-not (Test-Path $regPath)) {
+            New-Item -Path $regPath -Force | Out-Null
+          }
+
+          # Set the InstallPath value
+          Set-ItemProperty -Path $regPath -Name InstallPath -Value $installPath -Type String
+          Write-Host "Registry key set: $regPath -> InstallPath = $installPath"
+
+      - name: Move Replays and Maps to User Dir
+        shell: pwsh
+        run: |
+          # These files are expected in the user dir, so we move them here.
+
+          $source = "${{ inputs.gamedata }}\Replays"
+          $destination = "$env:USERPROFILE\Documents\Command and Conquer Generals Zero Hour Data\Replays"
+          Write-Host "Move replays to $destination"
+          New-Item -ItemType Directory -Path $destination -Force | Out-Null
+          Move-Item -Path "$source\*" -Destination $destination -Force
+
+          $source = "${{ inputs.gamedata }}\Maps"
```

Comment:
The naming of `inputs.gamedata` is ambiguous with GameData, used for the static game data. Maybe name it `inputs.userdata`.